### PR TITLE
Update page title for homepage

### LINKED ISSUE #1389: [amp.dev] Bad title for homepage
The homepage title is currently just "Start".

Also the link on the logo goes to `/index.html`. Please change to `/`.

### PR DESCRIPTION
**Pull request by matthiasrohmer**
This one fixes #1389 while at the same time introducing the functionality to define special titles for specific pages by using 

```yaml
---
$title: Start
$titles:
  title: AMP - a web component framework to easily create user-first web experiences
```

in their frontmatter. At the same time all page titles just defined by `$title` will be postfixed by " - AMP" just like on ampproject.org.

### CODE REVIEW

**Comment by sebastianbenz** [comment]
What's the difference between `title` and `titles`?

**Comment by matthiasrohmer** [comment]
The titles in the `$titles` dictionary can be called via `doc.titles(*key*)` which falls back to `$title` if no title with that key is defined. That allows to define different titles for various use cases (`<title>`, headline, in navigations). Currently we only made use of it for the header.

File: pages/content/amp-dev/index.html [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ---
 $title: Start
+$titles:
+  title: AMP - a web component framework to easily create user-first web experiences
```
Comment (sebastianbenz) [inline pages/content/amp-dev/index.html]: Why not replace `start` with the correct title?

Comment (matthiasrohmer) [reply pages/content/amp-dev/index.html]: Not the case but when the homepage would be met by any of the navigation walkers (sidebar, header) they would print this full-blown title.

We could flip the pattern and use `$titles.nav` instead for navigation titles.

**Comment by matthiasrohmer** [comment]
`$titles` is the required name from Grow. Only the keys inside the list are free to choose.

File: pages/content/amp-dev/index.html (hunk 1 above)
Comment (sebastianbenz) [reply pages/content/amp-dev/index.html]: > We could flip the pattern and use `$titles.nav` instead for navigation titles.

SGTM - that would make it more clear

**Comment by sebastianbenz** [comment]
Ah ok - wasn't aware. `titles` it is then. One comment, otherwise LGTM.

File: pages/content/amp-dev/index.html (hunk 1 above)
Comment (matthiasrohmer) [reply pages/content/amp-dev/index.html]: Cool, will ping you when done 🙂

**Comment by sebastianbenz** [comment]
@matthiasrohmer are you going to change to `$titles.nav` as part of this PR?

**Comment by matthiasrohmer** [comment]
In this one, I just want to wait for #1416.

**Comment by matthiasrohmer** [comment]
Closed by #1422 and #1424.

**Pull request closed without merging**